underfs/s3/src/main/java/tachyon/underfs/s3/S3UnderFileSystem.java

```diff
@@ -347,6 +333,23 @@ private boolean deleteInternal(String key) {
   }
 
   /**
+   * Gets the child name based on the parent name. If the parent is not valid, the entire child
```

Comment:
I agree about getChildName.
The naming of `stripPrefixIfPresent` seems to imply that its okay for the prefix not to be present. Currently its optional because we will call it on user inputs which may not contain the prefix (for example they may pass in paths relative to the bucket which should be allowed).